### PR TITLE
playercntl: Givecash improvements

### PR DESCRIPTION
**Pull request by lnvoker**
Main changes:
1) Now displays current time both for sender and receiver (offline too).
2) Added ability to attach commentary to /givecash

### CODE REVIEW

File: Plugins/Public/playercntl_plugin/Givecash.cpp [hunk 1]
```diff
@@ -413,19 +431,25 @@ namespace GiveCash
...
+			wstos(wscTargetCharname).c_str(), wstos(HkGetAccountID(HkGetAccountByCharname(wscTargetCharname))).c_str(),
+			wscComment.size() ? "| With comment:" : "", wstos(wscComment).c_str());
 
 		// A friendly message explaining the transfer.
-		msg = L"You have sent " + ToMoneyStr(cash) + L" credits to " + wscTargetCharname;
+		msg = time + L" You have sent " + ToMoneyStr(cash) + L" credits to " + wscTargetCharname;
 		if (bAnon)
 			msg += L" anonymously";
+		if (wscComment.size())
+			msg += L" with attached commentary";
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Givecash.cpp]:
```suggestion
			msg += L" with comment";
```

File: Plugins/Public/playercntl_plugin/Givecash.cpp [hunk 2]
```diff
@@ -401,9 +408,20 @@ namespace GiveCash
...
+		timeinfo = localtime(&rawtime);
+
+		char buffer[21];
+		strftime(buffer, 21, "[%Y-%m-%d]%H:%M:%S", timeinfo);
+		wstring time = stows((string)buffer);
+
 		// If the target player is online then send them a message saying
 		// telling them that they've received the cash.
-		wstring msg = L"You have received " + ToMoneyStr(cash) + L" credits from " + ((bAnon) ? L"anonymous" : wscCharname);
+		wstring msg = time + L" You have received " + ToMoneyStr(cash) + L" credits from " + ((bAnon) ? L"anonymous" : wscCharname) + (wscComment.size() ? L" | " + wscComment : L"");
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Givecash.cpp]:
```suggestion
		wstring msg = time + L" You have received " + ToMoneyStr(cash) + L" credits from " + ((bAnon) ? L"anonymous" : wscCharname) + (wscComment.size() ? L" | With comment: " + wscComment : L"");
```

File: Plugins/Public/playercntl_plugin/Main.cpp [hunk 3]
```diff
@@ -916,11 +916,11 @@ USERCMD UserCmds[] =
 	{ L"/set movecharcode",	Rename::UserCmd_SetMoveCharCode,	L"Usage: /set movecharcode <code>" },
 	{ L"/restart",		Restart::UserCmd_Restart,		L"Usage: /restart <faction>" },
 	{ L"/showrestarts",	Restart::UserCmd_ShowRestarts,	L"Usage: /showrestarts" },
-	{ L"/givecash",		GiveCash::UserCmd_GiveCash,		L"Usage: /givecash <charname> <cash> [anon] or /gc ..." },
-	{ L"/gc",			GiveCash::UserCmd_GiveCash,		L"Usage: /givecash <charname> <cash> [anon] or /gc ..." },
-	{ L"/givecasht",	GiveCash::UserCmd_GiveCashTarget,		L"Usage: /givecasht <cash> [anon] or /gct ..." },
-	{ L"/gct",			GiveCash::UserCmd_GiveCashTarget,		L"Usage: /givecasht <cash> [anon] or /gct ..." },
-	{ L"/sendcash",		GiveCash::UserCmd_GiveCash,		L"Usage: /givecash <charname> <cash> [anon]"},
+	{ L"/givecash",		GiveCash::UserCmd_GiveCash,		L"Usage: /givecash <charname> <cash> [anon] [commentary] or /gc ..." },
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Main.cpp]:
```suggestion
	{ L"/givecash",		GiveCash::UserCmd_GiveCash,		L"Usage: /givecash <charname> <cash> [anon] [comment] or /gc ..." },
```

File: Plugins/Public/playercntl_plugin/Main.cpp [hunk 4]
```diff
@@ -916,11 +916,11 @@ USERCMD UserCmds[] =
...
+	{ L"/gc",			GiveCash::UserCmd_GiveCash,		L"Usage: /givecash <charname> <cash> [anon] [commentary] or /gc ..." },
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Main.cpp]:
```suggestion
	{ L"/gc",			GiveCash::UserCmd_GiveCash,		L"Usage: /givecash <charname> <cash> [anon] [comment] or /gc ..." },
```

File: Plugins/Public/playercntl_plugin/Main.cpp [hunk 5]
```diff
@@ -916,11 +916,11 @@ USERCMD UserCmds[] =
...
+	{ L"/givecasht",	GiveCash::UserCmd_GiveCashTarget,		L"Usage: /givecasht <cash> [anon] [commentary] or /gct ..." },
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Main.cpp]:
```suggestion
	{ L"/givecasht",	GiveCash::UserCmd_GiveCashTarget,		L"Usage: /givecasht <cash> [anon] [comment] or /gct ..." },
```

File: Plugins/Public/playercntl_plugin/Main.cpp [hunk 6]
```diff
@@ -916,11 +916,11 @@ USERCMD UserCmds[] =
...
+	{ L"/gct",			GiveCash::UserCmd_GiveCashTarget,		L"Usage: /givecasht <cash> [anon] [commentary] or /gct ..." },
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Main.cpp]:
```suggestion
	{ L"/gct",			GiveCash::UserCmd_GiveCashTarget,		L"Usage: /givecasht <cash> [anon] [comment] or /gct ..." },
```

File: Plugins/Public/playercntl_plugin/Main.cpp [hunk 7]
```diff
@@ -916,11 +916,11 @@ USERCMD UserCmds[] =
...
+	{ L"/sendcash",		GiveCash::UserCmd_GiveCash,		L"Usage: /givecash <charname> <cash> [anon] [commentary]"},
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Main.cpp]:
```suggestion
	{ L"/sendcash",		GiveCash::UserCmd_GiveCash,		L"Usage: /givecash <charname> <cash> [anon] [comment]"},
```

**Comment by dsyalex** [comment]
Looks good and builds, let's get these string changes made and I'll test tomorrow when I'm less tired

File: Plugins/Public/playercntl_plugin/Givecash.cpp (hunk 2 above)
Comment (lnvoker) [reply Plugins/Public/playercntl_plugin/Givecash.cpp]: I think my version is better.

File: Plugins/Public/playercntl_plugin/Givecash.cpp [hunk 8]
```diff
@@ -413,19 +431,25 @@ namespace GiveCash
...
+			wstos(wscTargetCharname).c_str(), wstos(HkGetAccountID(HkGetAccountByCharname(wscTargetCharname))).c_str(),
+			wscComment.size() ? "| With comment:" : "", wstos(wscComment).c_str());
 
 		// A friendly message explaining the transfer.
-		msg = L"You have sent " + ToMoneyStr(cash) + L" credits to " + wscTargetCharname;
+		msg = time + L" You have sent " + ToMoneyStr(cash) + L" credits to " + wscTargetCharname;
 		if (bAnon)
 			msg += L" anonymously";
+		if (wscComment.size())
+			msg += L" with comment";
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Givecash.cpp]: colon, space, wscComment?

File: Plugins/Public/playercntl_plugin/Givecash.cpp [hunk 9]
```diff
@@ -401,9 +408,20 @@ namespace GiveCash
...
 
+		// Get current time
+		time_t rawtime;
+		tm* timeinfo;
+
+		time(&rawtime);
+		timeinfo = localtime(&rawtime);
+
+		char buffer[21];
+		strftime(buffer, 21, "[%Y-%m-%d]%H:%M:%S", timeinfo);
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Givecash.cpp]: space after ]

File: Plugins/Public/playercntl_plugin/Givecash.cpp (hunk 8 above)
Comment (lnvoker) [reply Plugins/Public/playercntl_plugin/Givecash.cpp]: But why do we need to duplicate comment? Commands you typing are being saved to chat.

Comment (dsyalex) [reply Plugins/Public/playercntl_plugin/Givecash.cpp]: okay, let's at least fix this though:
```suggestion
			msg += L" with a comment";
```

**Pull request merged**